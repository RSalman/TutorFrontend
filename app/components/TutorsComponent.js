import React, { Component } from 'react';
import { View, ListView, StyleSheet } from 'react-native';
import TutorRow from './TutorRow';

export default class TutorsComponent  extends Component {
  constructor(props) {
    super(props);
    const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 != r2});
    this.state = {
      dataSource: ds.cloneWithRows(this.getTutors())
    };
  }

  getTutors() {
    var testArray = [
      { name: "Sarmad", picture: "" },
      { name: "Mike", picture: "" },
      { name: "John", picture: "" },
      { name: "Dan", picture: "" },
      { name: "Test1", picture: "" },
      { name: "Test2", picture: "" },
      { name: "Test3", picture: "" },
      { name: "Test4", picture: "" },
      { name: "Salman", picture: "" },
      { name: "Muraad", picture: "" },
      { name: "Peng", picture: "" },
      { name: "Leila", picture: "" }
    ];
    return testArray;
  }

  getDataSource(tutors) {
    return this.state.dataSource.cloneWithRows(tutors);
  }

  render() {
    return (
      <ListView
       navigator={this.props.navigator}
        style={styles.container}
        dataSource={this.state.dataSource}
        renderRow={(row) => <TutorRow {...row}/>}
        renderSeparator={(sectionId, rowId) => <View key={rowId} style={styles.separator} />}
      />
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 10,
  },
  separator: {
    flex: 1,
    height: StyleSheet.hairlineWidth,
    backgroundColor: '#8E8E8E',
  },
});
