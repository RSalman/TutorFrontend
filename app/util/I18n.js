import I18n from 'react-native-i18n';

import en from './locales/en';
import fr from './locales/fr';

I18n.fallbacks = true;

I18n.translations = {
  en,
  fr
};

global.I18n = I18n;
