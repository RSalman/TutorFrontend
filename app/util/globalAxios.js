import axios from 'axios';
import { Alert, NetInfo } from 'react-native';

axios.defaults.baseURL = 'http://cpserver.eastus.cloudapp.azure.com:2000/api/v1';
axios.defaults.headers.post['Content-Type'] = 'application/json';
axios.defaults.headers.get['Content-Type'] = 'application/json';
axios.defaults.headers.post.Accept = 'application/json';
axios.defaults.headers.get.Accept = 'application/json';

var connection = true;
function connectivityChange(isConnected) {
  connection = isConnected;
}

function showNoInternetDialog(config) {
  return new Promise(function(resolve, reject) {
    Alert.alert(
      I18n.t('globalWifiDialog.noInternetConnection'),
      I18n.t('globalWifiDialog.fixInternet'),
      [
        { text: I18n.t('retry'), onPress: () => resolve(axios(config)) },
      ],
      { cancelable: false }
    );
  });
}

NetInfo.isConnected.addEventListener('change', connectivityChange);

axios.interceptors.response.use(undefined, function(error) {
  return connection ? Promise.reject(error) : showNoInternetDialog(error.config);
});

global.axios = axios;
