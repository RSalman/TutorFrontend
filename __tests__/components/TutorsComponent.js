import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import TutorsComponent from '../../app/components/TutorsComponent';
import '../../app/util/globalAxios';
import '../../app/util/I18n';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const dummyTutors = [
  {
    picture: '',
    rating: 5,
    degree: 'TEST1',
  },
  {
    picture: '',
    rating: 2,
    degree: 'TEST2',
  }
];

function setup(state) {
  const store = mockStore(state);
  const props = {
    updateTutors: jest.fn(),
    store
  };
  return props;
}

describe('<TutorsComponent />', () => {
  it('should render tutors in rows', () => {
    const props = setup({
      tutors: {
        allTutors: dummyTutors,
        isLoading: false
      }
    });
    const tree = renderer.create(
      <TutorsComponent {...props} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
