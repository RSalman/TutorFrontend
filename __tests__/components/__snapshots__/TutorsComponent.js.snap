exports[`<TutorsComponent /> should render tutors in rows 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "marginTop": 20,
    }
  }>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#393e42",
          "borderBottomColor": "#000",
          "borderBottomWidth": 1,
          "borderTopColor": "#000",
          "borderTopWidth": 1,
        },
        Object {
          "backgroundColor": "#e1e8ee",
          "borderBottomColor": "#e1e1e1",
          "borderTopColor": "#e1e1e1",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }>
    <TextInput
      autoCapitalize={undefined}
      autoCorrect={undefined}
      autoFocus={undefined}
      blurOnSubmit={undefined}
      clear={undefined}
      clearButtonMode={undefined}
      clearTextOnFocus={undefined}
      dataDetectorTypes={undefined}
      defaultValue={undefined}
      editable={undefined}
      enablesReturnKeyAutomatically={undefined}
      inlineImageLeft={undefined}
      inlineImagePadding={undefined}
      isFocused={undefined}
      keyboardAppearance={undefined}
      keyboardType={undefined}
      maxLength={undefined}
      multiline={undefined}
      numberOfLines={undefined}
      onBlur={undefined}
      onChange={undefined}
      onChangeText={[Function]}
      onContentSizeChange={undefined}
      onEndEditing={undefined}
      onFocus={undefined}
      onKeyPress={undefined}
      onLayout={undefined}
      onSelectionChange={undefined}
      onSubmitEditing={undefined}
      placeholder="Enter a course code or subject..."
      placeholderTextColor="#86939e"
      returnKeyLabel={undefined}
      returnKeyType={undefined}
      secureTextEntry={undefined}
      selectTextOnFocus={undefined}
      selectionColor="#86939e"
      selectionState={undefined}
      style={
        Array [
          Object {
            "backgroundColor": "#303337",
            "borderRadius": 3,
            "color": "#86939e",
            "fontSize": 17.5,
            "height": 30,
            "margin": 8,
            "overflow": "hidden",
            "paddingLeft": 26,
            "paddingRight": 19,
          },
          Object {
            "backgroundColor": "#bdc6cf",
          },
          false,
          Object {
            "borderRadius": 15,
          },
          Object {
            "paddingVertical": 0,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value={undefined} />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#86939e",
            "fontSize": 16,
          },
          Array [
            Object {
              "backgroundColor": "transparent",
              "left": 16,
              "position": "absolute",
              "top": 15.5,
            },
            undefined,
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }>
      
    </Text>
  </View>
  <View>
    <RCTScrollView>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "#ededed",
                  "borderBottomWidth": 1,
                  "paddingBottom": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }>
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 34,
                    "width": 34,
                  },
                  Object {
                    "borderRadius": 17,
                  },
                  Object {
                    "height": 50,
                    "width": 50,
                  },
                ]
              } />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }>
              <View
                style={undefined}>
                <View />
              </View>
              <View
                style={undefined}>
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "column",
                        "paddingLeft": 10,
                      }
                    }>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                      </View>
                      <View
                        style={undefined}>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "width": 150,
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "fontFamily": "HelveticaNeue-Thin",
                                },
                                undefined,
                              ]
                            }>
                             $
                            undefined
                            /hr 
                          </Text>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "fontFamily": "HelveticaNeue-Thin",
                                },
                                Object {
                                  "fontSize": 7,
                                },
                              ]
                            }>
                             (Avg. Rate)
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontFamily": "HelveticaNeue-Thin",
                            },
                            undefined,
                          ]
                        }>
                         
                        undefined
                         
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 0.15,
                  "justifyContent": "center",
                }
              }>
              <View
                onLongPress={undefined}
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white">
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc6cf",
                        "fontSize": 28,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          Object {},
                          undefined,
                        ],
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderBottomColor": "#ededed",
                  "borderBottomWidth": 1,
                  "paddingBottom": 10,
                  "paddingRight": 10,
                  "paddingTop": 10,
                },
                undefined,
              ],
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "marginLeft": 10,
                },
                undefined,
              ]
            }>
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 34,
                    "width": 34,
                  },
                  Object {
                    "borderRadius": 17,
                  },
                  Object {
                    "height": 50,
                    "width": 50,
                  },
                ]
              } />
            <View
              style={
                Object {
                  "flex": 1,
                  "justifyContent": "center",
                }
              }>
              <View
                style={undefined}>
                <View />
              </View>
              <View
                style={undefined}>
                <View>
                  <View
                    style={
                      Object {
                        "flexDirection": "column",
                        "paddingLeft": 10,
                      }
                    }>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }>
                      <View
                        style={
                          Object {
                            "flexDirection": "row",
                            "justifyContent": "space-between",
                          }
                        }>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                        <View
                          accessibilityComponentType="button"
                          accessibilityLabel={undefined}
                          accessibilityTraits="button"
                          accessible={true}
                          hitSlop={undefined}
                          isTVSelectable={true}
                          onLayout={undefined}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "opacity": 1,
                            }
                          }
                          testID={undefined}
                          tvParallaxProperties={undefined}>
                          <Text
                            accessible={true}
                            allowFontScaling={false}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "color": "gray",
                                  "fontFamily": "FontAwesome",
                                  "fontSize": 20,
                                  "fontStyle": "normal",
                                  "fontWeight": "normal",
                                },
                                Object {},
                              ]
                            }>
                            
                          </Text>
                        </View>
                      </View>
                      <View
                        style={undefined}>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "flexDirection": "column",
                              "justifyContent": "center",
                              "width": 150,
                            }
                          }>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "fontFamily": "HelveticaNeue-Thin",
                                },
                                undefined,
                              ]
                            }>
                             $
                            undefined
                            /hr 
                          </Text>
                          <Text
                            accessible={true}
                            allowFontScaling={true}
                            ellipsizeMode="tail"
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "transparent",
                                  "fontFamily": "HelveticaNeue-Thin",
                                },
                                Object {
                                  "fontSize": 7,
                                },
                              ]
                            }>
                             (Avg. Rate)
                          </Text>
                        </View>
                      </View>
                    </View>
                    <View
                      style={
                        Object {
                          "flex": 1,
                          "flexDirection": "row",
                        }
                      }>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "backgroundColor": "transparent",
                              "fontFamily": "HelveticaNeue-Thin",
                            },
                            undefined,
                          ]
                        }>
                         
                        undefined
                         
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flex": 0.15,
                  "justifyContent": "center",
                }
              }>
              <View
                onLongPress={undefined}
                onPress={undefined}
                style={
                  Array [
                    false,
                    false,
                    false,
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    },
                    undefined,
                  ]
                }
                underlayColor="white">
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#bdc6cf",
                        "fontSize": 28,
                      },
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Array [
                          Object {},
                          undefined,
                        ],
                      ],
                      Object {
                        "fontFamily": "Material Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }>
                  
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
